### PR TITLE
[braintree] fix subscription requests according to the docs

### PR DESCRIPTION
**Pull request by Qunapi**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://developers.braintreepayments.com/reference/request/subscription/create/node>

### CODE REVIEW

File: types/braintree/index.d.ts [hunk 1]
```diff
@@ -978,6 +976,16 @@ declare namespace braintree {
         trialPeriod?: boolean;
     }
 
+    export interface SubscriptionUpdateRequest extends SubscriptionRequest {
```
Comment (smrubin) [inline types/braintree/index.d.ts]: From what I'm seeing on the docs, the `id` and `merchantAccountId` as part of the update request object seem to be optional as well. I think we can keep the existing `SubscriptionRequest` interface and make both these fields optional on that interface.

**Review by smrubin — CHANGES_REQUESTED** [review]
@Qunapi Thanks for identifying this inconsistency with the docs. Per other comment, I think we should just mark the two fields as optional as part of the request object for both create and update. As a result, I don't think we need new interfaces for the create and update request objects as they'd be the same.

File: types/braintree/index.d.ts (hunk 1 above)
Comment (Qunapi) [reply types/braintree/index.d.ts]: But then we will have a slightly incorrect check for both update and create. Wouldn't that be a problem?

Comment (smrubin) [reply types/braintree/index.d.ts]: Not sure I follow, what do you mean by slightly incorrect check? The `id` is required for update but only as the first argument in the `update` method, not as part of the request object itself. The way I'm reading the docs, the request object marks both of the touched fields as optional.

Comment (sharno) [reply types/braintree/index.d.ts]: Actually it seems that `id` is optional in `create` but required in `update`
While `merchantAccountId` is optional in both of them

Comment (smrubin) [reply types/braintree/index.d.ts]: @sharno So, there are two separate `id`s we may be talking about here. There is an `id` passed as the first argument to the `update` method. This is required, and is currently marked as such in the TS definition.

The second `id` is passed as part of the request object, and per the docs says "If this option is provided", leading me to believe it is optional.

Note the differences between:
1) The id argument (https://developers.braintreepayments.com/reference/request/subscription/update/node#arg.id)
2) The id as part of the request object (https://developers.braintreepayments.com/reference/request/subscription/update/node#id)

Comment (sharno) [reply types/braintree/index.d.ts]: @smrubin Oh I see what you are referring to now. Your suggestion makes sense then of just making both fields optional in the current TS definition.

**Review by sharno — APPROVED** [review]
Looks good to me

File: types/braintree/index.d.ts [hunk 2]
```diff
@@ -976,6 +974,8 @@ declare namespace braintree {
         trialDuration?: number;
         trialDurationUnit?: string;
         trialPeriod?: boolean;
+        id?: string;
```
Comment (smrubin) [inline types/braintree/index.d.ts]: /nit Can we keep the props in alphabetical order? We've done this throughout and I think it makes lookups easier.

**Review by smrubin — APPROVED** [review]
Awesome, thanks @Qunapi

**Pull request merged**